### PR TITLE
[bluray] Blu-ray meta data retrival

### PR DESCRIPTION
**Pull request by ace20022**

### CODE REVIEW

File: xbmc/filesystem/BlurayDirectory.cpp [hunk 1]
```diff
@@ -201,4 +201,102 @@ CURL CBlurayDirectory::GetUnderlyingCURL(const CURL& url)
...
+  }
+
+  m_dll->bd_set_debug_handler(DllLibbluray::bluray_logger);
+  m_dll->bd_set_debug_mask(DBG_CRIT | DBG_BLURAY | DBG_NAV);
+
+  m_bd = m_dll->bd_init();
+
+  if (!m_bd)
+  {
+    CLog::Log(LOGERROR, "CDVDInputStreamBluray::Open - failed to initialize libbluray");
```
Comment (ksooo) [inline xbmc/filesystem/BlurayDirectory.cpp]: CDVDInputStreamBluray::Open => CBlurayDirectory::InitializeBluray or use `__FUNCTION__`

File: xbmc/filesystem/BlurayDirectory.cpp [hunk 2]
```diff
@@ -201,4 +201,102 @@ CURL CBlurayDirectory::GetUnderlyingCURL(const CURL& url)
...
+    return false;
+  }
+
+  std::string langCode;
+  g_LangCodeExpander.ConvertToISO6392T(g_langInfo.GetDVDMenuLanguage(), langCode);
+  m_dll->bd_set_player_setting_str(m_bd, BLURAY_PLAYER_SETTING_MENU_LANG, langCode.c_str());
+  
+  if (!m_dll->bd_open_files(m_bd, &root, DllLibbluray::dir_open, DllLibbluray::file_open))
+  {
+    CLog::Log(LOGERROR, "CDVDInputStreamBluray::Open - failed to open %s", CURL::GetRedacted(root).c_str());
```
Comment (ksooo) [inline xbmc/filesystem/BlurayDirectory.cpp]: CDVDInputStreamBluray::Open => CBlurayDirectory::InitializeBluray or use `__FUNCTION__`

File: xbmc/filesystem/BlurayDirectory.cpp [hunk 3]
```diff
@@ -201,4 +201,102 @@ CURL CBlurayDirectory::GetUnderlyingCURL(const CURL& url)
   return CURL(host.append(filename));
 }
 
+bool CBlurayDirectory::InitializeBluray(std::string &root)
+{
+  m_dll = new DllLibbluray();
+  if (!m_dll->Load())
+  {
+    CLog::Log(LOGERROR, "CBlurayDirectory::GetDirectory - failed to load dll");
```
Comment (ksooo) [inline xbmc/filesystem/BlurayDirectory.cpp]: CBlurayDirectory::GetDirectory => CBlurayDirectory::InitializeBluray or use `__FUNCTION__`

File: xbmc/filesystem/BlurayDirectory.cpp [hunk 4]
```diff
@@ -201,4 +201,102 @@ CURL CBlurayDirectory::GetUnderlyingCURL(const CURL& url)
...
+std::string CBlurayDirectory::HexToString(const uint8_t *buf, int count)
+{
+  std::array<char, 42> tmp;
+
+  for (int i = 0; i < count; i++) {
+    sprintf(tmp.data() + (i * 2), "%02x", buf[i]);
+  }
+
+  std::string id(std::begin(tmp), std::end(tmp));
+  return id;
```
Comment (ksooo) [inline xbmc/filesystem/BlurayDirectory.cpp]: maybe just `return std::string(std::begin(tmp), std::end(tmp));`?

File: xbmc/filesystem/BlurayDirectory.cpp [hunk 5]
```diff
@@ -201,4 +201,102 @@ CURL CBlurayDirectory::GetUnderlyingCURL(const CURL& url)
...
+}
+
+std::string CBlurayDirectory::GetBlurayID()
+{
+  if (!m_blurayInitialized)
+    return "";
+
+  const BLURAY_DISC_INFO* disc_info = m_dll->bd_get_disc_info(m_bd);
+  if (!disc_info || !disc_info->bluray_detected)
+    return "";
```
Comment (ksooo) [inline xbmc/filesystem/BlurayDirectory.cpp]: Seeing the above six lines verbatim in `CBlurayDirectory::GetBlurayTitle()`. MAybe you shoul factor them out into a separate method?

File: xbmc/storage/MediaManager.cpp [hunk 6]
```diff
@@ -529,34 +530,51 @@ std::string CMediaManager::GetDiskUniqueId(const std::string& devicePath)
...
       StringUtils::StartsWith(pathVideoTS, "iso9660://"))
-    pathVideoTS = g_mediaManager.TranslateDevicePath(""); 
+      pathVideoTS = g_mediaManager.TranslateDevicePath("");
+
+    CLog::Log(LOGDEBUG, "GetDiskUniqueId: Trying to retrieve ID for path %s", pathVideoTS.c_str());
 
-  CLog::Log(LOGDEBUG, "GetDiskUniqueId: Trying to retrieve ID for path %s", pathVideoTS.c_str());
 
+    CFileItem item(pathVideoTS, false);
+    CDVDInputStreamNavigator dvdNavigator(NULL, item);
```
Comment (ksooo) [inline xbmc/storage/MediaManager.cpp]: NULL => nullptr ?

File: xbmc/filesystem/BlurayDirectory.cpp [hunk 7]
```diff
@@ -154,12 +156,10 @@ bool CBlurayDirectory::GetDirectory(const CURL& url, CFileItemList &items)
...
 
-  m_dll = new DllLibbluray();
-  if (!m_dll->Load())
-  {
-    CLog::Log(LOGERROR, "CBlurayDirectory::GetDirectory - failed to load dll");
+  if (!InitializeBluray(root))
     return false;
-  }
+
+  uint8_t* pic = GetThumbnail();
```
Comment (Rechi) [inline xbmc/filesystem/BlurayDirectory.cpp]: pic is unused

File: xbmc/filesystem/BlurayDirectory.cpp [hunk 8]
```diff
@@ -201,4 +201,102 @@ CURL CBlurayDirectory::GetUnderlyingCURL(const CURL& url)
...
+
+uint8_t* CBlurayDirectory::GetThumbnail()
+{
+  if (!m_blurayInitialized)
+    return nullptr;
+
+  const struct meta_dl* meta = m_dll->bd_get_meta(m_bd);
+  uint8_t* picture = nullptr;
+  int64_t size = 0;
+  int suc = m_dll->bd_get_meta_file(m_bd, meta->thumbnails->path, &picture, &size);
```
Comment (Rechi) [inline xbmc/filesystem/BlurayDirectory.cpp]: suc is unused

File: xbmc/filesystem/BlurayDirectory.cpp [hunk 9]
```diff
@@ -201,4 +201,102 @@ CURL CBlurayDirectory::GetUnderlyingCURL(const CURL& url)
...
+}
```
Comment (Rechi) [inline xbmc/filesystem/BlurayDirectory.cpp]: this function may not return anything, but return type isn't void

**Review by Rechi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**